src/game/Chat/Level2.cpp

```diff
@@ -2317,6 +2317,165 @@ bool ChatHandler::HandleNpcShowLootCommand(char* /*args*/)
+        else if (creature->GetFormationSlot())
+            PSendSysMessage("Creature is in formation slot");
+        else
+            SendSysMessage("Error: unable to retrieve the slot of the creature.");
+    }
+
+    SendSysMessage(gData->to_string().c_str());
+    return true;
+}
+
+// show detailed information of group behavior
+// bool ChatHandler::HandleNpcGroupBehaviorShowCommand(char* /*args*/)
+// {
+//     Creature* creature = getSelectedCreature();
+// 
+//     if (!creature)
+//     {
+//         SendSysMessage(LANG_SELECT_CREATURE);
+//         SetSentErrorMessage(true);
+//         return false;
+//     }
+// 
+// 
+//     return true;
+// }
+
+// set behavior of targeted group
+// bool ChatHandler::HandleNpcGroupBehaviorSetCommand(char* args)
+// {
+//     Creature* creature = getSelectedCreature();
+// 
+//     if (!creature)
+//     {
+//         SendSysMessage(LANG_SELECT_CREATURE);
+//         SetSentErrorMessage(true);
+//         return false;
+//     }
+// 
+//     return true;
+// }
+
+// reset creature formation template
+bool ChatHandler::HandleNpcFormationInfoCommand(char* /*args*/)
+{
+    Creature* creature = getSelectedCreature();
+
+    if (!creature)
+    {
+        SendSysMessage(LANG_SELECT_CREATURE);
+        SetSentErrorMessage(true);
+        return false;
+    }
+
+    auto currSlot = creature->GetFormationSlot();
+    if (!currSlot)
+    {
+        SendSysMessage("Creature is not in formation");
+        return true;
+    }
+
+    SendSysMessage(currSlot->GetFormationData()->to_string().c_str());
+    return true;
+}
+
+// reset creature formation template
+bool ChatHandler::HandleNpcFormationResetCommand(char* /*args*/)
+{
+    Creature* creature = getSelectedCreature();
+
+    if (!creature)
+    {
+        SendSysMessage(LANG_SELECT_CREATURE);
+        SetSentErrorMessage(true);
+        return false;
+    }
+
+    auto currSlot = creature->GetFormationSlot();
+    if (!currSlot)
+    {
+        SendSysMessage("Creature is not in formation");
+        return true;
+    }
+
+    currSlot->GetFormationData()->Reset();
+
+    // need implementation
+    PSendSysMessage("Formation is reset to default!");
+    return true;
+}
+
+// change creature formation template
+bool ChatHandler::HandleNpcFormationSwitchCommand(char* args)
+{
+    Creature* creature = getSelectedCreature();
+
+    if (!creature)
+    {
+        SendSysMessage(LANG_SELECT_CREATURE);
+        SetSentErrorMessage(true);
+        return false;
+    }
+
+    auto currSlot = creature->GetFormationSlot();
+    if (!currSlot)
+    {
+        SendSysMessage("Creature is not in formation");
+        return true;
+    }
+
+    uint32 formationId;
+    if (!ExtractUInt32(&args, formationId))
+    {
+        PSendSysMessage("Please provide a valid formation id!\n.npc formation switch #formationId");
+        return false;
+    }
+
+    if (formationId >= static_cast<uint32>(SpawnGroupFormationType::SPAWN_GROUP_FORMATION_TYPE_COUNT))
```

Comment:
primitives are best typecasted as c-style cast, since they arent really heresy. Curious choice of enum class, shauren loves them, buuuut historically in cmangos they seem out of place. Not really opposed to it tho.